scripts/train.py

```diff
@@ -45,6 +45,11 @@ def create_models(num_classes, weights='imagenet', multi_gpu=0):
     # Keras recommends initialising a multi-gpu model on the CPU to ease weight sharing, and to prevent OOM errors.
     # optionally wrap in a parallel model
     if args.multi_gpu > 1:
+
+        if args.batch_size < args.multi_gpu:
```

Comment:
What about batch size though? That's not an argument to `create_models`. Should the check maybe be placed in `main()` where access to `args` is fine then?

Reply:
@de-vri-es I originally wanted it in that location, but it seemed more appropriate to place it in context for the multi-gpu parallelism, I wanted to avoid bloating the main. As below, I have messaged Hans on Slack for further guidance on whether we should modify the signature, or how to treat args as a whole for these decisions.

Reply:
I'm fine with checking consistency of arguments in main. The earlier we catch errors in the arguments the better.

Suppose that we would create a directory for the snapshots or perform some other actions, we don't want to catch the errors *after* we did that (it might leave unused empty directories).

Currently it isn't much of an issue since nothing is really happening before we create the model, but in general checking arguments for errors should be the first step of a command line program, I think.

Reply:
Hans recommended I talk to you regarding placing the batch_size check within the main. I proposed a utility function to wrap parse_args, which would pick out inherent contradictions in settings. Also then denotes a place for any future constraints to be placed; thus avoiding main clutter.

The return of parse_args function which is currently `return parser.parse_args()` could be `return check_args(parser.parse_args())`. Whereby inherent faulty settings are immediately detected and thrown back prior to backend initialisation ( which takes a while for 8 GPUs ). Speeds up time to feedback on settings, alongside the above comments.

Reply:
Sounds good. No clutter and early fault detection :)

👍